RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate NotificationPreferencesView to hooks

## Proposed changes

Migrate NotificationPreferencesView to hooks and improves layout and user feedback 
- It is no longer necessary to change screens to select some options, they are via actionSheet.
- Now when an error occurs when changing the option, it reverts to the old option.

## Issue(s)	


## How to test or reproduce

- Try changing some option and see if it will actually replicate on the web or in tests.
- Force an error (too many requests for example) and see the error message and the go back option.

## Screenshots

## Types of changes




## Checklist



## Further comments
